src/QueryEditor.tsx

```diff
@@ -20,7 +20,7 @@ export const QueryEditor: React.FC<Props> = ({ app, query, datasource, onChange,
   };
 
   const handleColumnChange = (items: Array<SelectableValue<string>>) => {
-    const columns = items.map(({ value, dataType }) => ({ name: value!, dataType }));
+    const columns = items.map(({ value, dataType, columnType }) => ({ name: value!, dataType, columnType }));
```

Comment:
We now store the `columnType` in the query so we know if a column is the index or not.